src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexLexer.cs

```diff
@@ -0,0 +1,448 @@
+                    WorkspacesResources.Capture_group_numbers_must_be_less_than_or_equal_to_Int32_MaxValue,
+                    token.GetSpan()));
+            }
+
+            return token;
+        }
+
+        public RegexToken? TryScanCaptureName()
+        {
+            if (Position == Text.Length)
+            {
+                return null;
+            }
+
+            var start = Position;
+            while (Position < Text.Length && RegexCharClass.IsWordChar(this.CurrentChar))
+            {
+                Position++;
+            }
+
+            if (Position == start)
+            {
+                return null;
+            }
+
+            var token = CreateToken(RegexKind.CaptureNameToken, ImmutableArray<RegexTrivia>.Empty, GetSubPattern(start, Position));
+            token = token.With(value: token.VirtualChars.CreateString());
+            return token;
+        }
+
+        public RegexToken? TryScanNumberOrCaptureName()
+            => TryScanNumber() ?? TryScanCaptureName();
+
+        public RegexToken? TryScanOptions()
+        {
+            var start = Position;
+            while (Position < Text.Length && IsOptionChar(this.CurrentChar))
+            {
+                Position++;
+            }
+
+            return start == Position
+                ? default(RegexToken?)
+                : CreateToken(RegexKind.OptionsToken, ImmutableArray<RegexTrivia>.Empty, GetSubPattern(start, Position));
+        }
+
+        private bool IsOptionChar(char ch)
+        {
+            switch (ch)
+            {
+                case '+': case '-':
+                case 'i': case 'I':
+                case 'm': case 'M':
+                case 'n': case 'N':
+                case 's': case 'S':
+                case 'x': case 'X':
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public RegexToken ScanHexCharacters(int count)
+        {
+            var start = Position;
+            var beforeSlash = start - 2;
+
+            // Make sure we're right after the \x or \u.
+            Debug.Assert(Text[beforeSlash].Char == '\\');
+            Debug.Assert(Text[beforeSlash + 1].Char == 'x' || Text[beforeSlash + 1].Char == 'u');
+
+            for (int i = 0; i < count; i++)
+            {
+                if (Position < Text.Length && IsHexChar(this.CurrentChar))
+                {
+                    Position++;
+                }
+            }
+
+            var result = CreateToken(
+                RegexKind.TextToken, ImmutableArray<RegexTrivia>.Empty, GetSubPattern(start, Position));
+
+            var length = Position - start;
+            if (length != count)
+            {
+                result = result.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                    WorkspacesResources.Insufficient_hexadecimal_digits,
+                    TextSpan.FromBounds(Text[beforeSlash].Span.Start, Text[Position - 1].Span.End)));
```

Comment:
GetTextSpan(beforeSlash, Position)?

Reply:
I think this comment is just to say that we have a helper method that could be used here.
Something to consider.

Reply:
Indeed.  Updating to that.

Reply:
Fixed.